torchvision/prototype/transforms/functional/_color.py

```diff
@@ -211,7 +211,34 @@ def solarize(inpt: features.InputTypeJIT, threshold: float) -> features.InputTyp
         return solarize_image_pil(inpt, threshold=threshold)
 
 
-autocontrast_image_tensor = _FT.autocontrast
+def autocontrast_image_tensor(image: torch.Tensor) -> torch.Tensor:
+
+    if not (isinstance(image, torch.Tensor)):
+        raise TypeError("Input img should be Tensor image")
+
+    c = get_num_channels_image_tensor(image)
+
+    if c not in [1, 3]:
+        raise TypeError(f"Input image tensor permitted channel values are {[1, 3]}, but found {c}")
+
+    if image.numel() == 0:
+        # exit earlier on empty images
+        return image
+
+    bound = 1.0 if image.is_floating_point() else 255.0
+    dtype = image.dtype if torch.is_floating_point(image) else torch.float32
+
+    minimum = image.amin(dim=(-2, -1), keepdim=True).to(dtype)
+    maximum = image.amax(dim=(-2, -1), keepdim=True).to(dtype)
+
+    scale = bound / (maximum - minimum)
+    eq_idxs = maximum == minimum
```

Comment:
Reading back to original code that I wrote, I see I've used `eq_idxs = torch.isfinite(scale).logical_not()` for the same thing. To be honest, I don't understand why I did that (it looks kind of stupid). Just to be safe, you tested this logic with special cases where the values are the same across different channels and the result is identical right?

From code I expect that your approach is faster and better. Just being a bit paranoid.

Reply:
Thanks for asking, I'll check with more attention to that

Reply:
I did few other checks and I think `eq_idxs = maximum == minimum` should be equivalent to `eq_idxs = torch.isfinite(scale).logical_not()`. So, we can keep PR's implementation.